tests/dom/basictests.cpp

```diff
@@ -308,13 +308,49 @@ namespace number_tests {
   }
 
+  bool json_index_for_number() {
+    simdjson::dom::parser parser;
+    const auto string_result = parser.parse("\"string\""_padded).get_location_of_number_in_json();
+    if (!string_result.error()) {
+      std::cerr << "Expected error, but failed" << std::endl;
+      return false;
+    }
+    const auto root_result_1 = parser.parse("1234"_padded).get_location_of_number_in_json();
+    if (root_result_1.error()) { std::cerr << root_result_1.error() << std::endl; return false; }
+    if (root_result_1.value_unsafe() != 0) {
+      std::cerr << "Expected 0, but got " << root_result_1.value_unsafe() << std::endl;
+      return false;
+    }
+    const auto root_result_2 = parser.parse(" 1234"_padded).get_location_of_number_in_json();
+    if (root_result_2.error()) { std::cerr << root_result_2.error() << std::endl; return false; }
+    if (root_result_2.value_unsafe() != 1) {
+      std::cerr << "Expected 1, but got " << root_result_2.value_unsafe() << std::endl;
+      return false;
+    }
+    const auto root_result_3 = parser.parse("-1234"_padded).get_location_of_number_in_json();
+    if (root_result_3.error()) { std::cerr << root_result_3.error() << std::endl; return false; }
+    if (root_result_3.value_unsafe() != 0) {
+      std::cerr << "Expected 1, but got " << root_result_3.value_unsafe() << std::endl;
+      return false;
+    }
+    const auto element = parser.parse("{\"a\":1234}"_padded);
```

Comment:
Thoughts on more test cases that could be added:

Add tests of uint64, e.g. 18446744073709551615 or change some tests to use those?
Add tests of double, e.g. 1.0

(in case of changes to the tape format unintentionally breaking this)